### DISCUSSION TITLE
Using BrowserStack for Cross Browser Compatibility

### DISCUSSION THREAD

**Discussion by SudharakaP** [opencollective/opencollective#4319] [Ideas]
Just a random idea; I don't know much about the stats we have about the number of bugs we get due to dependencies/differences in different browsers. But I was wondering if it's worthwhile to integrate Cypress tests with [BrowserStack](https://www.browserstack.com/) so that we might be able to increase our cross browser compatibility. Is this something of interest? 🐨 🐼 

BrowserStack seems to also have an OpenSource program which I think we can apply; https://www.browserstack.com/open-source

@opencollective/core-developers @opencollective/core

**Comment by Betree** [comment]
Browserstack offers many services, are you referring to the ability to add screenshot testing/reviews? Or do you meant to run cypress tests with Browserstack to run them on multiple browsers at once?

I've been experimenting with https://www.chromatic.com lately on this topic (it's easy to plug now that we have a storybook). It's super interesting to get these visual diffs that you need to approve whenever there's a UI change to prevent regressions.

**Reply by SudharakaP** [reply]
> Browserstack offers many services, are you referring to the ability to add screenshot testing/reviews? Or do you meant to run cypress tests with Browserstack to run them on multiple browsers at once?
> 
> I've been experimenting with https://www.chromatic.com lately on this topic (it's easy to plug now that we have a storybook). It's super interesting to get these visual diffs that you need to approve whenever there's a UI change to prevent regressions.

@Betree : My initial thought was to run cypress tests with Browserstack on multiple browsers to make sure we are really supporting all major browsers so that we can catch the Firefox issues and stuff that we get sometimes. 😄 

Ah, didn't know about Chromatic. Yes the visual diffs will be super helpful. 😄

**Reply by Betree** [reply]
That can be interesting, though [we may not need](https://www.cypress.io/blog/2020/02/06/introducing-firefox-and-edge-support-in-cypress-4-0/) to use Browserstack to achieve that. I also think that the errors that we could spot with this strategy in fact pretty rare - I mainly see things like event handling for inputs or popover menus (eg. https://github.com/opencollective/opencollective/issues/3826), things that we don't properly test today. They're pretty rare in part because we don't support really difficult browsers (like IE), but also because the technologies we rely on with JS often handle cross-browser compatibility for us. For example, `styled-components` automatically prefix the CSS rules for major browsers. 

After that, the most common issues I've seen on Open Collective regarding cross-browser are related to styling: something looking bigger on some browsers, an image or a mask not being displayed properly (like in the collective page Hero), CSS flex not behaving the same way in Chrome than Firefox, that kind of stuff. Cross-browser cypress tests won't help for that.

**Reply by SudharakaP** [reply]
@Betree : Thanks for the input. Let us part this idea for the moment then. 🙂
